pkg/ddl/ddl_worker.go

```diff
@@ -637,41 +637,25 @@ func cleanMDLInfo(pool *sess.Pool, job *model.Job, ec *clientv3.Client, ownerID
 	} else {
 		sql = fmt.Sprintf("delete from mysql.tidb_mdl_info where job_id = %d and owner_id = '%s'", job.ID, ownerID)
 	}
-	sctx, _ := pool.Get()
-	defer pool.Put(sctx)
+	sctx, _ := s.sessPool.Get()
+	defer s.sessPool.Put(sctx)
 	se := sess.NewSession(sctx)
 	se.GetSessionVars().SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlmostFull)
-	_, err := se.Execute(context.Background(), sql, "delete-mdl-info")
+	_, err := se.Execute(s.schCtx, sql, "delete-mdl-info")
 	if err != nil {
 		logutil.DDLLogger().Warn("unexpected error when clean mdl info", zap.Int64("job ID", job.ID), zap.Error(err))
 		return
 	}
-	if cleanETCD && ec != nil {
+	// TODO do we need clean it for JobStateRollbackDone?
```

Comment:
`JobStateRollbackDone` does trigger schema version changes for some DDLs, if we don't clean keys here, there may be a leak?

Reply:
yes, that's why i want to add a JobStateRollbackSynced, so we can check it easily